File: server/core/src/main/scala/org/infinispan/server/core/transport/IdleStateHandlerProvider.scala
package org.infinispan.server.core.transport

import io.netty.channel.{ChannelHandlerContext, ChannelInboundHandlerAdapter}
import io.netty.handler.timeout.IdleStateEvent

/**
 * A Netty channel handler that allows idle channels to be closed.
 *
 * @author Galder Zamarreño
 * @since 4.1
 */
class IdleStateHandlerProvider extends ChannelInboundHandlerAdapter {


  override def userEventTriggered(ctx: ChannelHandlerContext, evt: scala.Any): Unit = {
    System.out.println(evt)
    if (evt.isInstanceOf[IdleStateEvent]) {
      ctx.close
    }
    ctx.fireUserEventTriggered(evt)
  }
}

File: server/core/src/main/java/org/infinispan/server/core/transport/IdleStateHandlerProvider.java
package org.infinispan.server.core.transport;

import io.netty.channel.ChannelHandlerContext;
import io.netty.channel.ChannelInboundHandlerAdapter;
import io.netty.handler.timeout.IdleStateEvent;

/**
 * A Netty channel handler that allows idle channels to be closed.
 *
 * @author Galder Zamarreño
 * @author wburns
 * @since 4.1
 */
public class IdleStateHandlerProvider extends ChannelInboundHandlerAdapter {

   @Override
   public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
      if (evt instanceof IdleStateEvent) {
         ctx.close();
      }
      ctx.fireUserEventTriggered(evt);
   }
}